### PR TITLE
#1545 child for admins

### PR DESCRIPTION
**Pull request by LeraKornienko**

### CODE REVIEW

File: src/app/shared/enum/enumUA/tech-admin/users-tabs.ts [hunk 1]
```diff
@@ -14,4 +14,6 @@ export enum UserTabs {
   all,
   parent,
   child
-}
\ No newline at end of file
+}
+
+
```
Comment (litvinets) [inline src/app/shared/enum/enumUA/tech-admin/users-tabs.ts:19]: revert these chanegs

**Review by litvinets — CHANGES_REQUESTED** [review]
image
1) wrong table view
2) search does not work
3)

File: src/app/shared/services/children/children.service.ts [hunk 2]
```diff
@@ -20,19 +21,31 @@ export class ChildrenService {
     private store: Store,
   ) { }
 
-  private setParams( searchString?:string ): HttpParams {
+  private setParams( parameters?: ChildrenParameters, isParent?: boolean ): HttpParams {
     let params = new HttpParams();
 
-    if(searchString){
-      params = params.set('SearchString', searchString);
+    if(parameters){
```
Comment (sivxkxn) [inline src/app/shared/services/children/children.service.ts:27]: formatting in the file

File: src/app/shared/services/children/children.service.ts [hunk 3]
```diff
@@ -20,19 +21,31 @@ export class ChildrenService {
...
+      if(parameters.searchString){
+      params = params.set('SearchString', parameters.searchString);}
+      if(parameters.tabTitle){ 
+        if(parameters.tabTitle == "Батьки"){
```
Comment (sivxkxn) [inline src/app/shared/services/children/children.service.ts:31]: === instead of ==

File: src/app/shared/services/children/children.service.ts [hunk 4]
```diff
@@ -20,19 +21,31 @@ export class ChildrenService {
...
+          isParent= true;
+        }
+        if(parameters.tabTitle == "Діти"){
```
Comment (sivxkxn) [inline src/app/shared/services/children/children.service.ts:34]: the same here

File: src/app/shared/services/children/children.service.ts [hunk 5]
```diff
@@ -61,8 +74,8 @@ export class ChildrenService {
     /**
    * This method get children for Admin
    */
-  getChildrenForAdmin(searchString: string): Observable<ChildCards> {
-    const options = { params: this.setParams(searchString) };
+  getChildrenForAdmin(paremeters, isParent?): Observable<ChildCards> {
```
Comment (sivxkxn) [inline src/app/shared/services/children/children.service.ts:77]: please change spelling on "parameters" and add types for arguments

File: src/app/shared/services/children/children.service.ts [hunk 6]
```diff
@@ -61,8 +74,8 @@ export class ChildrenService {
...
+    const options = { params: this.setParams(paremeters, isParent), };
```
Comment (sivxkxn) [inline src/app/shared/services/children/children.service.ts:78]: also spelling here

File: src/app/shared/services/children/children.service.ts [hunk 7]
```diff
@@ -20,19 +21,31 @@ export class ChildrenService {
...
+      if(parameters.tabTitle){ 
+        if(parameters.tabTitle == "Батьки"){
+          isParent= true;
+        }
+        if(parameters.tabTitle == "Діти"){
+          isParent= false;
+        }
+      }
+
+    if(isParent !== undefined){
```
Comment (sivxkxn) [inline src/app/shared/services/children/children.service.ts:39]: I'm not sure about this check. I guess isParent should be false by default. @litvinets what do you think?

**Review by litvinets — APPROVED** [review]

**Pull request merged**